src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -0,0 +1,375 @@
+                    "Unknown project id '{projectId}' in collection '{collectionId}' reported in notification",
+                    buildEvent.ResourceContainers.Project.Id,
+                    buildEvent.ResourceContainers.Collection.Id
+                );
+                return NoContent();
+            }
+
+            _logger.LogInformation("Resolved build '{buildId}' for project '{project}, fetching build details",
+                buildEvent.Resource.Id,
+                projectName);
+            Build build = await Client.GetBuild(projectName, buildEvent.Resource.Id);
+            if (IsIgnoredReason(build))
+            {
+                return NoContent();
+            }
+
+            if (IsIgnoredStatus(build))
+            {
+                return NoContent();
+            }
+
+            await ProcessBuildNotificationsAsync(build);
+
+            return NoContent();
+        }
+
+        private bool IsIgnoredReason(Build build)
+        {
+            switch (build.Reason)
+            {
+                case "batchedCI":
+                case "buildCompletion":
+                case "individualCI":
+                case "manual":
+                case "schedule":
+                case "scheduleForced":
+                case "triggered":
+                case "userCreated":
+                    _logger.LogInformation("Reason '{reason}' detected for build '{buildId}'", build.Reason, build.Id);
+                    return false;
+                case "pullRequest":
+                case "checkInShelveset":
+                case "validateShelveset":
+                    _logger.LogDebug("Ignored reason code '{reason}' found", build.Reason);
+                    return true;
+                default:
+                    _logger.LogWarning("Undocumented reason code '{reason}' found", build.Reason);
+                    return true;
+            }
+        }
+
+        private bool IsIgnoredStatus(Build build)
+        {
+            switch (build.Result)
+            {
+                case "failed":
+                case "partiallySucceeded":
+                    _logger.LogInformation("Result '{result}' detected for build '{buildId}'", build.Result, build.Id);
+                    return false;
+                case "canceled":
+                case "succeeded":
+                    _logger.LogDebug("Ignored result code '{result}' found", build.Result);
+                    return true;
+                default:
+                    _logger.LogWarning("Undocumented result code '{result}' found", build.Result);
+                    return true;
+            }
+        }
+
+        private async Task ProcessBuildNotificationsAsync(Build build)
+        {
+            const string fullBranchPrefix = "refs/heads/";
+            foreach (var monitor in _options.Value.Monitor.Builds)
+            {
+                if (!string.Equals(build.Project.Name, monitor.Project, StringComparison.OrdinalIgnoreCase))
```

Comment:
missing "{" and "}" in the next two code blocks. Also, perhaps they can be joined.

Reply:
They are really long and complicated. Joining all 3 of these if's together would create one thing that's a nightmare to read.

But I will add the braces.